src/search.c

```diff
@@ -1543,8 +1544,6 @@ static GString *get_grep_options(void)
 		g_string_append_c(gstr, 'i');
 	if (settings.fif_match_whole_word)
 		g_string_append_c(gstr, 'w');
-	if (settings.fif_recursive)
-		g_string_append_c(gstr, 'r');
```

Comment:
Non-recursive search for "test" in geany/ with none/all files specified:

"/bin/grep: .git: Is a directory
.gitignore:113:# /tests/
<more git ignores>
<real matches>
/bin/grep: data: Is a directory
/bin/grep: doc: Is a directory
<more directories>"

They are a fix, and I didn't wait to make two conflicting patches.